src/Http/Controllers/FittingController.php

```diff
@@ -179,12 +179,16 @@ public function getFittingList()
         foreach ($fittings as $fit) {
             $ship = InvType::where('typeName', $fit->shiptype)->first();
 
-            array_push($fitnames, [
-                'id' => $fit->id,
-                'shiptype' => $fit->shiptype,
-                'fitname' => $fit->fitname,
-                'typeID' => $ship->typeID
-            ]);
+            if ($ship && property_exists($ship, 'typeID')) {
+                array_push($fitnames, [
+                    'id' => $fit->id,
+                    'shiptype' => $fit->shiptype,
+                    'fitname' => $fit->fitname,
+                    'typeID' => $ship->typeID
+                ]);
+            } else {
+                $this->deleteFittingById($fit->id);
```

Comment:
This might be harsh, just my opinion of how to deal with it.

Reply:
I would not delete it in this case.... The reason being is that you will see the same error if a user does not have an SDE installed, and this will wipe all of their fits. I would just log an error with the message stating something to the effect of fit id x contains a non valid ship name